sdk/storage/azure-storage-file-share/tests/test_directory.py

```diff
@@ -1396,6 +1399,53 @@ def test_bad_audience_directory_client(self, **kwargs):
         # Assert
         directory_client.exists()
 
+    @FileSharePreparer()
+    @recorded_by_proxy
+    def test_file_permission_format_for_directory_operations(self, **kwargs):
+        storage_account_name = kwargs.pop("storage_account_name")
+        storage_account_key = kwargs.pop("storage_account_key")
+
+        self._setup(storage_account_name, storage_account_key)
+        share_client = self.fsc.get_share_client(self.share_name)
+        directory_client = share_client.create_directory('dir1')
+
+        directory_properties_on_creation = directory_client.get_directory_properties()
+        last_write_time = directory_properties_on_creation.last_write_time
+        creation_time = directory_properties_on_creation.creation_time
+        change_time = directory_properties_on_creation.change_time
+
+        new_last_write_time = last_write_time + timedelta(hours=1)
+        new_creation_time = creation_time + timedelta(hours=1)
+        new_change_time = change_time + timedelta(hours=1)
+        expected_permission_key = '8984294407537026961*15810287295243203168'
+
+        directory_client.set_http_headers(
+            file_attributes='None',
+            file_creation_time=new_creation_time,
+            file_last_write_time=new_last_write_time,
+            file_change_time=new_change_time,
+            file_permission=TEST_FILE_PERMISSIONS_IN_SDDL,
+            file_permission_format="SDDL"
+        )
+
+        props = directory_client.get_directory_properties()
+        assert props is not None
+        assert props.creation_time == new_creation_time
+        assert props.last_write_time == new_last_write_time
+        assert props.change_time == new_change_time
+        assert props.permission_key == expected_permission_key
```

Comment:
Permission key is not the same as permission, it will also be different for every run of this test I would guess. You should be checking permissions here and below. This would also apply to the file test.

Reply:
I had to refresh my understanding here so to get the permission, you will need to take this permission key and call `get_permission_for_share` at the ShareClient level to see what the actual permission is. Then you can compare it to the one you set.